community/bolt/src/test/java/org/neo4j/bolt/v1/transport/integration/Neo4jWithSocket.java

```diff
@@ -26,38 +26,48 @@
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.function.Consumer;
+import java.util.function.Supplier;
 
 import org.neo4j.bolt.BoltKernelExtension;
 import org.neo4j.graphdb.GraphDatabaseService;
 import org.neo4j.graphdb.config.Setting;
+import org.neo4j.io.fs.FileSystemAbstraction;
 import org.neo4j.test.TestGraphDatabaseFactory;
 
 import static org.neo4j.graphdb.factory.GraphDatabaseSettings.BoltConnector.EncryptionLevel.OPTIONAL;
 import static org.neo4j.graphdb.factory.GraphDatabaseSettings.boltConnector;
 
 public class Neo4jWithSocket implements TestRule
 {
+    private Supplier<FileSystemAbstraction> fileSystemProvider;
     private final Consumer<Map<Setting<?>,String>> configure;
-    TestGraphDatabaseFactory graphDatabaseFactory;
+    private TestGraphDatabaseFactory graphDatabaseFactory;
     private GraphDatabaseService gdb;
 
     public Neo4jWithSocket()
     {
-        this( new TestGraphDatabaseFactory(), settings -> {} );
+        this( new TestGraphDatabaseFactory(), () -> null, settings -> {} );
     }
 
     public Neo4jWithSocket( Consumer<Map<Setting<?>, String>> configure )
     {
-        this( new TestGraphDatabaseFactory(), configure );
+        this( new TestGraphDatabaseFactory(), () -> null, configure );
     }
 
     public Neo4jWithSocket( TestGraphDatabaseFactory graphDatabaseFactory, Consumer<Map<Setting<?>, String>> configure )
     {
+        this( graphDatabaseFactory, () -> null, configure );
```

Comment:
can we have a proper default here? What does `null` give you? "Ephemeral" or "real"?